File: Dynamic Programming/clears.cpp
//Odrediti minimalnu stranicu matrice koja je simetricna i horizontalno i vertikalno
//u kojoj se jedinice ne dodiruju, i sadrzi zadati broj jedinica
#include <stdio.h>
int n,r;
int main()
 {
  scanf("%d",&n);
  if (n!=3)
  {
    for (r=1; r*r+(r-1)*(r-1)<n; r++);
    printf("%d\n",r+r-1);
  } else puts("5");
  return 0;
}

File: Dynamic Programming/clears.pas
{Odrediti minimalnu stranicu matrice koja je simetricna i horizontalno i vertikalno
 u kojoj se jedinice ne dodiruju, i sadrzi zadati broj jedinica}
program p1;
var i,n:longint;
begin
  read(n);
  if n=3 then writeln(5) else
  begin
    i:=1;
    while (i*i+1) div 2<n do inc(i,2);
    writeln(i);
  end;
end.
